### ISSUE TITLE
Akka-actor-typed has a broken sources jar

### ISSUE THREAD

**Issue by ahrooran-r** [akka/akka#31040]
```
  <dependency>
            <groupId>com.typesafe.akka</groupId>
            <artifactId>akka-actor-typed_${scala.binary.version}</artifactId>
            <version>${akka.version}</version>
  </dependency>
```
 
I use intellij idea and used this jar in pom. When I clicked download sources, it downloads the sources jar but fails to properly show the documentations and still shows bytecode.

So I tried to manually download the sources using maven and manually selected source files in intellij. Still doesn't work.

IDEA is correctly working because it shows documentation for other jars. 

Please look into it

**Comment by raboof** [comment]
> IDEA is correctly working because it shows documentation for other jars

Do you have an example of another Scala dependency jar for which this works? Does it work for other Akka jars or only for other jars from different projects?

**Comment by ahrooran-r** [comment]
It does not work for any akka jars. I tried on following version as well:
        ```
<!-- https://mvnrepository.com/artifact/com.typesafe.akka/akka-actor-typed -->
        <dependency>
            <groupId>com.typesafe.akka</groupId>
            <artifactId>akka-actor-typed_2.13</artifactId>
            <version>2.5.32</version>
        </dependency>

```

But it downloads the sources jar for all versions but fails to unpack and display it.

For dependencies from different projects such as logback, lombok are working fine. IDEA correctly downloads and displays the sources.

**Comment by raboof** [comment]
> It does not work for any akka jars. I tried on following version as well

Makes sense - thanks for testing on 2.5.32 as well!

> For dependencies from different projects such as logback, lombok are working fine

That makes sense. logback and lombok are Java projects though - could you try with another Scala project?

Unfortunately I can't reproduce the problem myself yet: I see the sources just fine, but it could of course be that I configured something for that at some point. I'm using IntelliJ 2021.3.1 CE (Build #IC-213.6461.79, built on December 28, 2021) and version 2021.3.17 of the Scala plugin.

Are you seeing this in a Java or a Scala project? Do you have the Scala plugin installed? It is possible that that is required to see the sources, since Akka is implemented largely in Scala.

**Comment by ahrooran-r** [comment]
Before installing Scala plugin the situation is this (as I have mentioned before):

![Screenshot 2022-01-07 001308](https://user-images.githubusercontent.com/46846338/148435834-6907e950-e869-44b2-960b-c11f914b5f2d.png)

![Screenshot 2022-01-07 001329](https://user-images.githubusercontent.com/46846338/148435845-0ac05c16-435b-4762-b4b8-e965565909f1.png)

I could not see the sources of jar.

Then as you said, I installed Scala plugin (I haven't installed this before). But now another weird problem occurs:

![Screenshot 2022-01-07 001537](https://user-images.githubusercontent.com/46846338/148436080-53fd8bd1-a57b-4760-9e36-378951707333.png)

![Screenshot 2022-01-07 001952](https://user-images.githubusercontent.com/46846338/148436086-22b81bde-3593-4aa2-b63a-b89b358ac515.png)

I can see the sources file. But intellij underlines everything and says it could not find the files. But the jars are all in external libraries folder. I manually opened one scala file in idea through navigating the jars and it shows the sources jar. 

So I don't know what causes this. I use jdk 17 though? would that be the culprit? But akka works well with jdk 17.

IntelliJ IDEA 2021.3.1 (Community Edition)
Build #IC-213.6461.79, built on December 28, 2021
Scala Build 2021.3.17

Java version 
openjdk 17 2021-09-14
OpenJDK Runtime Environment (build 17+35-2724)
OpenJDK 64-Bit Server VM (build 17+35-2724, mixed mode, sharing)

**Comment by raboof** [comment]
> I installed Scala plugin (I haven't installed this before). I can see the sources file.

OK, so you need the Scala plugin to view Scala sources - that makes sense I guess?

> But intellij underlines everything and says it could not find the files

Hmm, so installing the Scala plugin somehow interfered with your Java project building properly? That's very weird.

> I use jdk 17 though? would that be the culprit? But akka works well with jdk 17.

I can't really imagine how that could cause this.

Does cleaning the project or re-importing the pom.xml help?

**Comment by ahrooran-r** [comment]
Nope. I cleared everything and started again but it does not work

**Comment by johanandren** [comment]
If you look at the contents of the source jar in your local maven cache, using for example something like `jar tf ~/.m2/repository/com/typesafe/akka/akka-actor_2.13/2.6.18/akka-actor_2.13-2.6.18-sources.jar` does that work?

It should give you a list of 257 files.

**Comment by raboof** [comment]
> Nope. I cleared everything and started again but it does not work

Strange. Perhaps something to discuss over at https://intellij-support.jetbrains.com/hc/en-us/community/topics/200381545-Scala ?

**Comment by ahrooran-r** [comment]
SOLVED:

Apparantly I used scala 3 version which the IntelliJ Scala plugin does not support.

So after seeing your version I changed **scala version to 2.13**. Now both functionalities are working well.

After a bit of googling, I found that Scala 3 support has just been introduced to Scala plugin. Hence it is not stable and has some bugs.

In mvn repo, **akka_actor_2.13** is the go to. Although it has support for version 3, intellij support is not yet complete and for java users it does not matter.

Thanks for working out with me on this issue by the way. :)

**Comment by johanandren** [comment]
Sounds about right, that means the jar is probably correct but something else with IntelliJ not finding the right sources for types. 

I noticed in the Java Akka quickstart project, changed to Scala 3, with the source jars available from my local cache (`mvn dependency:sources`) I had to build the project before IntelliJ found them.

I saw you answered and got it solved while I was writing this. Good 👍
